src/main/java/uk/gov/hmcts/reform/rolecategorymigration/Application.java

```diff
@@ -0,0 +1,127 @@
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cloud.openfeign.EnableFeignClients;
+import org.springframework.context.ApplicationContext;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ForkJoinPool;
+import java.util.regex.Pattern;
+
+@EnableFeignClients(basePackages = {"uk.gov.hmcts.reform.idam"})
+@SpringBootApplication
+@Slf4j
+public class Application implements CommandLineRunner {
+
+	private final List<Pattern> roleCategoryPatterns = new ArrayList<>();
+	private final String ExceptionLabel = "*EXCEPTION*";
+
+	@Autowired
+	private ApplicationContext context;
+
+	@Autowired
+	private CaseDataRepository caseDataRepository;
+
+	@Autowired
+	private IdamRepository idamRepository;
+
+	public static void main(String[] args) {
+		SpringApplication.run(Application.class, args);
+	}
+
+	@Override
+	public void run(String... args) {
+		log.info("Data migration task has begun");
+		long startTime = System.currentTimeMillis();
+		roleCategoryPatterns.add(RoleCategory.CITIZEN.getPattern());
+		roleCategoryPatterns.add(RoleCategory.JUDICIAL.getPattern());
+		roleCategoryPatterns.add(RoleCategory.PROFESSIONAL.getPattern());
+
+		log.info("Starting to populate null role categories");
+
+		populateNewRoleCategory();
+
+		long endTime = System.currentTimeMillis();
+
+		log.info("Completed data population task in " + (endTime - startTime) + " milliseconds");
+
+		SpringApplication.exit(context, () -> 0);
+	}
+
+	private void populateNewRoleCategory() {
+
+		Page<String> dataWithoutRoleCategory = caseDataRepository.findCaseUsersById(PageRequest.of(0, 5000));
+		if (dataWithoutRoleCategory.isEmpty()) {
+			return;
+		}
+		List<String> listOfUserIds = new ArrayList<>(dataWithoutRoleCategory.getContent());
+		Map<String, List<String>> rolesToUserIdMap = new ConcurrentHashMap<>();
+		ForkJoinPool customThreadPool = new ForkJoinPool(10);
+		try {
+			customThreadPool.submit(() -> listOfUserIds.parallelStream().forEach(userId -> {
+				try {
+					rolesToUserIdMap.put(userId, retrieveIdamRoles(userId));
+				} catch (Exception exception) {
+					caseDataRepository.updateRoleCategory(ExceptionLabel, userId);
+					log.error(exception.getMessage());
+				}
+			})).get();
+		} catch (Exception exception) {
+			log.error(exception.getMessage());
+		} finally {
+			customThreadPool.shutdown();
+		}
+
+		RoleCategory matchingCategory;
+
+		for (String userId : rolesToUserIdMap.keySet()) {
+			try {
+				matchingCategory = matchRoleCategory(rolesToUserIdMap.get(userId), userId);
```

Comment:
```suggestion
				RoleCategory matchingCategory = matchRoleCategory(rolesToUserIdMap.get(userId), userId);
```
We can set it as a local variable in here instead of on L84